src/taipy/gui/data/pandas_data_accessor.py

```diff
@@ -74,13 +74,25 @@ def __build_transferred_cols(
             data = data.copy()
             is_copied = True
             for k, v in styles.items():
-                applied = False
+                col_applied = False
                 func = gui._get_user_function(v)
                 if callable(func):
-                    applied = self.__apply_user_function(gui, func, k if k in cols else None, v, data)
-                if not applied:
+                    col_applied = self.__apply_user_function(gui, func, k if k in cols else None, v, data, "tps__")
+                if not col_applied:
                     data[v] = v
-                cols.append(v)
+                cols.append(col_applied or v)
+        if tooltips:
+            # copy the df so that we don't "mess" with the user's data
+            if not is_copied:
+                # copy the df so that we don't "mess" with the user's data
+                data = data.copy()
```

Comment:
No performance issue I guess? Just copying references?

Reply:
not copying reference but we never modify the user's data